File: app/src/main/java/com/example/firabasefirstexperience/Adapter/UserAdapter.java
package com.example.firabasefirstexperience.Adapter;

import android.content.Context;
import android.content.Intent;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.example.firabasefirstexperience.Activities.ChatActivity;
import com.example.firabasefirstexperience.Model.User;
import com.example.firabasefirstexperience.R;


import java.util.List;

public class UserAdapter extends RecyclerView.Adapter<UserAdapter.ViewHolder> {

    private Context mContext;
    private List<User> listUsers;

    public UserAdapter(Context mContext, List<User> listUsers) {
        this.mContext = mContext;
        this.listUsers = listUsers;
    }


    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.user_item, viewGroup, false);
        return new UserAdapter.ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int i) {
        final User user = listUsers.get(i);
        viewHolder.tv_Name.setText(user.getUserName());


        viewHolder.itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(mContext, ChatActivity.class);
                intent.putExtra("userId", user.getId());
                mContext.startActivity(intent);
            }
        });

    }

    @Override
    public int getItemCount() {
        return listUsers.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {

        public TextView tv_Name;
        public ImageView userImage;


        public ViewHolder(View itemView) {
            super(itemView);

            tv_Name = itemView.findViewById(R.id.tv_Name);
            userImage = itemView.findViewById(R.id.userImage);
        }
    }


}

File: app/src/main/java/com/nassdk/supchat/presentation/searchusers/UserAdapter.kt
package com.nassdk.supchat.presentation.searchusers

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.nassdk.supchat.R
import com.nassdk.supchat.model.User
import com.nassdk.supchat.presentation.diffprofile.ui.DiffProfileActivity
import de.hdodenhof.circleimageview.CircleImageView

class UserAdapter(private var listUsers: List<User>) : RecyclerView.Adapter<UserAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.user_item, parent, false)
        return ViewHolder(view)
    }

    override fun getItemCount(): Int {
        return listUsers.size
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.bind(listUsers[position])
    }


    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        private var tv_Name: TextView? = itemView.findViewById(R.id.tv_Name)
        private var userImage: CircleImageView? = itemView.findViewById(R.id.userImage)

        fun bind(user: User) {
            tv_Name?.text = user.userName

            if (user.imageURL == "default") {
                userImage?.setImageResource(R.mipmap.ic_launcher_round)
            } else {
                Glide
                        .with(itemView.context)
                        .load(user.imageURL)
                        .into(userImage!!)
            }

            itemView.setOnClickListener {
                val intent = Intent(itemView.context, DiffProfileActivity::class.java)
                intent.putExtra("id", user.id)
                itemView.context.startActivity(intent)
            }
        }
    }
}